src/main/java/by/itech/lab/supplier/service/impl/CustomerServiceImpl.java

```diff
@@ -9,24 +12,70 @@
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 import org.springframework.stereotype.Service;
+import javax.transaction.Transactional;
+import java.util.Optional;
 
 @Service
 @AllArgsConstructor
 public class CustomerServiceImpl implements CustomerService {
 
     private final CustomerRepository customerRepository;
     private final CustomerMapper customerMapper;
+    private final UserServiceImpl userService;
 
     @Override
-    public Page<CustomerDto> getCustomers(final Pageable pageable, final Boolean status) {
+    public Page<CustomerDto> findAllByActive(final Pageable pageable, final Boolean status) {
         return customerRepository.findByStatus(pageable, status).map(customerMapper::mapToCustomerView);
     }
 
     @Override
-    public CustomerDto getCustomer(final Long customerId) {
+    public CustomerDto findById(final Long customerId) {
         return customerRepository.findById(customerId)
                 .map(customerMapper::mapToCustomerView)
                 .orElseThrow(() -> new ResourceNotFoundException("Customer with id=" + customerId + " doesn't exist"));
     }
 
+    @Override
+    public CustomerDto save(CustomerDto customerDto) {
+        Customer customer = Optional.ofNullable(customerDto.getId())
+                .map(item -> {
+                    final Customer existing = customerRepository
+                            .findById(customerDto.getId())
+                            .orElseThrow();
+                    updateCustomer(customerDto, existing);
+                    return existing;
+                })
+                .orElseGet(() -> customerMapper.map(customerDto));
+
+        final Customer saved = customerRepository.save(customer);
+        userService.save(createAdmin(customerDto));
+        return customerMapper.map(saved);
+    }
+
+    private UserDto createAdmin(CustomerDto customerDto) {
```

Comment:
what's the correct method order?

Reply:
maybe better to put this logic to UserService?as here you exactly doing a job which related only to that service

Reply:
Maybe, we discuss it tomorrow!